specs/Deployment/DeploymentAPI.md

```diff
@@ -0,0 +1,136 @@
+expected to only have the App package and frameworks, and the WAS main package may not be present,
+or may not be at the desired version. This API enables a developer to detect when the required WAS
+main and/or framework packages are not present and take corrective action.
+
+Since this API is part of the WAS framework, the framework itself must be present.
+For MSIX packages, the framework must be listed as a dependency in the application manifest and
+will not be deployable without a min version being present. 
+
+Developers must not assume that the WAS main package is present, or that the version
+present is the minimum version required to support the feature set they expect. Instead, this API
+should be used to validate the WAS is the correct version and all required packages are present.
+
+The developer scenarios for this API is to answer the questions:
+1. Does this system have the min version of the Windows App SDK I require?
+2. What do I do to resolve this if it doesn't?
+
+The API addresses #1 by a simple query of the version they seek and a possible check for OS
+updates. #2 has two possible answers. First is direct resolution by the developer by acquiring and
+installing the packages if possible. Since the main packages are small and the target developer
+scenario will have permissions to deploy them, we will minimize developer and user pain by
+including the main package MSIX packages within the framework, and this API will allow for the
+deployment of that package if needed.
+
+For this release, packages without permission to deploy are not supported, and future versions
+will have solutions for such packages.
+
+# Examples
+
+## GetStatus
+
+This is the means by which an app or program can call to verify the runtime packages is present to
+a minimum version needed by the loaded framework. It should be called once after activation before
+trying to use the Windows App Runtime. This method returns a WindowsAppRuntime object
+that describes the current deployment state of the Windows App Runtime.
+
+Developers call this method without parameters, as all information about the version, channel, and
+architecture are derived from the current framework loaded and we assume any main packages needed
+by those frameworks are included within those framework packages.
+
+This method is good to identify if there is possible work required in the form of package install
+which may take several seconds to complete.
+
+```C++
+auto status =
+Windows::ApplicationModel::WindowsAppRuntime::Deployment::GetStatus();
+if (!status.IsOK())
+{
+    ...
+}
+```
+
+## Initialize
+
+Does a status check and if any packages are required for install, attempts to register those packages
+and get the WindowsAppRuntime into an OK state. This will perform package install if possible. The
+returned WindowsAppRuntimeStatus will contain the first error if one occurs during package install.
```

Comment:
This implies Initialize may do multiple things in a best-effort fashion and only return at the end returning 1st error. True?

If Initialize could, say, install 2 packages (e.g. WARmain + WARsingleton) and fails on the 1st, will it also try to install the 2nd and in the end return the 1st error encountered? This seems implied but not clearly stated.